Theme: Restructure Color System using Semantic Organization

shahinbashar opened the pull request:
### Overview
This PR refactors our theming system to implement a semantic color organization approach. We've moved from color names that describe their appearance (e.g., crayolaBlue) to names that describe their purpose in the UI (e.g., primary, error). This makes our theming system more maintainable and easier to customize.

### Changes
- Completely reorganized our color system using semantic naming and purpose-driven structure
- Implemented a modular file structure using part/part of directives for better code organization
- Added semantic status colors (success, error, warning, info, etc.)
- Fixed constant constructor issues by replacing getters with static constants
- Implemented proper copyWith() and lerp() methods for all theme extensions
- Split component-specific colors into separate files under the part directory

### Key Improvements
1. Semantic Color Organization
Colors are now organized by their purpose in the UI rather than by appearance:
```
// OLD: Based on appearance
static const Color crayolaBlue = Color(0xFF1877F2);

// NEW: Based on purpose
static const Color primary = _Colors.blue;
static const Color error = _Colors.red;
```
2. Better Component Isolation
Component-specific colors like AppBar, BottomNavBar, and PageView are now isolated in their own files, making the code more maintainable.
3. Proper Theme Transitions
The lerp() methods now properly handle color interpolation, enabling smooth theme transitions.
4. Improved Theme Customization
The copyWith() methods support all properties, making it easier to create theme variants.
5. Status Colors
Added semantic status colors that communicate meaning:
```
static const Color success = _Colors.green;
static const Color error = _Colors.red;
static const Color warning = _Colors.yellow;
static const Color info = _Colors.blue;
```
### Technical Notes

- Used static constants instead of getters to fix issues with constant constructors
- Implemented the ColorExtension interface for both light and dark themes
- Components follow a consistent pattern for accessing theme colors

shahinbashar committed:
feat: implement dark theme

momshaddinury commented on lib/src/core/service/router/router_state/router_state_provider.dart:
```diff
@@ -36,6 +36,9 @@ class RouterState extends _$RouterState {
       state = Routes.onboarding;
+      // Mark onboarding as completed
+      local?.save(CacheKey.isOnBoardingCompleted, true);
+      // Return the onboarding page
```
This comment is unnecessary. Remove it.

momshaddinury commented on lib/src/core/theme/src/theme_data.dart:
Remove `darkColor` from `$LightThemeData` and remove `color` from `$DarkThemeData`

momshaddinury commented on lib/src/core/theme/src/part/app_bar_theme.dart:
Rename `_AppBarTheme` to `_AppBarLightTheme`

momshaddinury commented on lib/src/core/theme/src/theme_extensions/extensions.dart:
```diff
@@ -1,10 +1,10 @@
 
 mixin ThemeExtensions {
-  final ColorExtension color = const ColorExtension();
+  final LightColorExtension color = const LightColorExtension();
```
Rename `color` to `lightColor`

dinurymomshad committed:
chore: Addressed PR feedbacks

dinurymomshad committed:
refactor: restructure color system to use semantic organization

- Reorganize colors based on their purpose rather than appearance
- Implement proper theme color inheritance with static constants
- Add semantic status colors (success, error, warning, info)
- Split implementation into modular files using part/part of directives
- Implement functional copyWith() and lerp() methods for easier theme customization
- Fix constant constructor issues by using static const fields

This change improves maintainability by making color roles explicit,
simplifies creation of theme variants, and enables smooth theme transitions
through proper color interpolation.
